File: Serene/Serene.Script/Imports/ServiceContracts/Northwind.SalesByCategoryService.cs
namespace Serene.Northwind
{
    using jQueryApi;
    using Serenity;
    using System;
    using System.Collections;
    using System.Collections.Generic;
    using System.Runtime.CompilerServices;

    [Imported, PreserveMemberCase]
    public partial class SalesByCategoryService
    {
        [InlineConstant] public const string BaseUrl = "Northwind/SalesByCategory";
    
        [InlineCode("Q.serviceRequest('Northwind/SalesByCategory/List', {request}, {onSuccess}, {options})")]
        public static jQueryXmlHttpRequest List(ListRequest request, Action<ListResponse<SalesByCategoryRow>> onSuccess, ServiceCallOptions options = null)
        {
            return null;
        }
    
        [Imported, PreserveMemberCase]
        public static class Methods
        {
            [InlineConstant] public const string List = "Northwind/SalesByCategory/List";
        }
    }
    
}

File: Serene/Serene.Core/Imports/ServerTypings/Northwind.SalesByCategoryService.ts
namespace Serene.Northwind {
    export namespace SalesByCategoryService {
        export const baseUrl = 'Northwind/SalesByCategory';

        export declare function List(request: Serenity.ListRequest, onSuccess?: (response: Serenity.ListResponse<SalesByCategoryRow>) => void, opt?: Q.ServiceOptions<any>): JQueryXHR;

        export declare const enum Methods {
            List = "Northwind/SalesByCategory/List"
        }

        [
            'List'
        ].forEach(x => {
            (<any>SalesByCategoryService)[x] = function (r, s, o) {
                return Q.serviceRequest(baseUrl + '/' + x, r, s, o);
            };
        });
    }
}
